sentry-android-core/src/main/java/io/sentry/android/core/internal/tombstone/TombstoneParser.java

```diff
@@ -0,0 +1,195 @@
+package io.sentry.android.core.internal.tombstone;
+
+import androidx.annotation.NonNull;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.protocol.DebugImage;
+import io.sentry.protocol.DebugMeta;
+import io.sentry.protocol.Mechanism;
+import io.sentry.protocol.Message;
+import io.sentry.protocol.SentryException;
+import io.sentry.protocol.SentryStackFrame;
+import io.sentry.protocol.SentryStackTrace;
+import io.sentry.protocol.SentryThread;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+public class TombstoneParser {
+
+  private final InputStream tombstoneStream;
+  private final Map<String, String> excTypeValueMap = new HashMap<>();
+
+  public TombstoneParser(InputStream tombstoneStream) {
```

Comment:
We usually are very keen on marking as much as possible as `final` as well as annotating any non-primitive method parameters as `@Nullable` or `@NotNull`
```suggestion
  public TombstoneParser(final @NotNull InputStream tombstoneStream) {
```